rescan.go

```diff
@@ -773,12 +788,12 @@ rescanLoop:
 					return fmt.Errorf("unable to register "+
 						"block subscription: %v", err)
 				}
-				defer func() {
-					if blockSubscription != nil {
-						blockSubscription.Cancel()
-						blockSubscription = nil
-					}
-				}()
+
+				// Clear any retry blocks as we are now
+				// considered current with the chain.
+				if len(retryBlocks) > 0 {
+					retryBlocks = retryBlocks[:0]
```

Comment:
If we get to this point: we're synced, but still have retry blocks, has an assumption in the current code been violated?

Reply:
If we have N retry blocks and N-1 causes us to no longer be current, then we'll need to clear these once we become current again.

Other than when an explicit rescan rewind is done, the only way for us to no longer be current is for our notification source to dispatch a notification out of order. This ideally shouldn't happen unless there's a bug in that logic.